src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/TimedFlipFlopDetector.java

```diff
@@ -0,0 +1,114 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.collections.TimeExpiringSet;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Impact;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A {@link FlipFlopDetector} whose recorded actions expire after a given period of time.
+ *
+ * <p>This class defines a flip flop as an {@link Impact#DECREASES_PRESSURE}s followed by an
+ * {@link Impact#INCREASES_PRESSURE}s to be a flip flops.
+ *
+ * <p>This class stores a {@link TimeExpiringSet} of {@link ImpactVector}s per {@link NodeKey}
+ * that are used to determine these flip flops.
+ */
+public class TimedFlipFlopDetector implements FlipFlopDetector {
```

Comment:
Nice, like how you keep this extensible.

Reply:
+1.